App/Sagas/PhotoViewingSagas.ts

```diff
@@ -1,20 +1,37 @@
-import { call, put, select } from 'redux-saga/effects'
-import { ActionType } from 'typesafe-actions'
+import { delay } from 'redux-saga'
+import { call, put, select, take } from 'redux-saga/effects'
+import { ActionType, getType } from 'typesafe-actions'
 
 import PhotoViewingActions, { ThreadData } from '../Redux/PhotoViewingRedux'
-import { photoAndComment } from '../Redux/PhotoViewingSelectors'
-import UIActions from '../Redux/UIRedux'
+import { photoAndComment, shouldNavigateToNewThread, photoToShareToNewThread } from '../Redux/PhotoViewingSelectors'
 import TextileNode from '../../TextileNode'
-import { Threads, Thread, Photo, BlockId } from '../Models/TextileTypes'
+import { Threads, Photo, BlockId, PhotoId } from '../Models/TextileTypes'
 import NavigationService from '../Services/NavigationService'
+import { shareWalletImage } from './ImageSharingSagas'
+
+export function * monitorNewThreadActions () {
+  while (true) {
+    const action: ActionType<typeof PhotoViewingActions.threadAdded> = yield take(getType(PhotoViewingActions.threadAdded))
+    const shouldNav: boolean = yield select(shouldNavigateToNewThread)
+    const photoToShare: { threadName: string, imageId: PhotoId, comment?: string} | undefined = yield select(photoToShareToNewThread)
+    yield put(PhotoViewingActions.clearNewThreadActions())
+    const { id, name } = action.payload
+    if (photoToShare && photoToShare.threadName === name) {
+      const { imageId, comment } = photoToShare
+      yield call(shareWalletImage, imageId, id, comment)
+    }
+    if (shouldNav) {
+      yield put(PhotoViewingActions.viewThread(action.payload.id))
+      yield delay(700)
```

Comment:
This delay is to work around that bouncy nav bug we have when creating a new thread.